### ISSUE TITLE
[clang-tidy[request] Use init list for container instead of (multiple) push_back(), emplace_back() etc.

### ISSUE THREAD

**Issue by chrchr-github** [llvm/llvm-project#117872]
~~~c++
#include <vector>

int f() {
    std::vector<int> v;
    v.push_back(1);
    v.push_back(2);
    v.push_back(3); // warn
    return v.back();
}

int g() {
    std::vector<int> v{ 1, 2, 3 }; // compliant
    return v.back();
}
~~~

**Comment by firewave** [comment]
This is #46672 with a different solution offered.

**Comment by denzor200** [comment]
> This is [#46672](https://github.com/llvm/llvm-project/issues/46672) with a different solution offered.

This issue regarding new check in **modernize** section, I suppose. The issue you've mentioned about **performance** one

**Comment by denzor200** [comment]
@EugeneZelenko should we add check-request label here? I don't know any existent check which we can extend to handle this issue

**Comment by denzor200** [comment]
This future check also should catch `clear` before multiple `push_back`.

BEFORE:
```
int f() {
    std::vector<int> v;
    v.push_back(1);
    v.push_back(2);
    v.push_back(3);
    v.clear();
    v.push_back(4);
    v.push_back(5);
    v.push_back(6);
    return v.back();
}
```

AFTER:
```
int f() {
    std::vector<int> v{1, 2, 3};
    v = {4, 5, 6};
    return v.back();
}
```

**Comment by denzor200** [comment]
This is a special case of another checker: https://github.com/llvm/llvm-project/issues/25337
